src/weights.jl

```diff
@@ -2,19 +2,21 @@
 abstract type AbstractWeights{S<:Real, T<:Real, V<:AbstractVector{T}} <: AbstractVector{T} end
 
 """
-    @weights name
+    @weights name [ctor=true]
```

Comment:
I'd probably just use `constructor` since it's easier to parse and we're only typing it in a couple places.